src/events/attributes.rs

```diff
@@ -331,6 +331,95 @@ impl<'a> From<(&'a str, &'a str)> for Attribute<'a> {
     }
 }
 
+// 1) All line breaks MUST have been normalized on input to #xA as described in 2.11 End-of-Line Handling, so the rest of this algorithm operates on text normalized in this way.
+// 2) Begin with a normalized value consisting of the empty string.
+// 3) For each character, entity reference, or character reference in the unnormalized attribute value, beginning with the first and continuing to the last, do the following:
+//   * For a character reference, append the referenced character to the normalized value.
+//   * For an entity reference, recursively apply step 3 of this algorithm to the replacement text of the entity.
+//   * For a white space character (#x20, #xD, #xA, #x9), append a space character (#x20) to the normalized value.
+//   * For another character, append the character to the normalized value.
+//
+// If the attribute type is not CDATA, then the XML processor MUST further process the normalized attribute value by discarding any leading and trailing space (#x20) characters,
+// and by replacing sequences of space (#x20) characters by a single space (#x20) character.
+//
+// Note that if the unnormalized attribute value contains a character reference to a white space character other than space (#x20), the normalized value contains the referenced
+// character itself (#xD, #xA or #x9). This contrasts with the case where the unnormalized value contains a white space character (not a reference), which is replaced with a
+// space character (#x20) in the normalized value and also contrasts with the case where the unnormalized value contains an entity reference whose replacement text contains a
+// white space character; being recursively processed, the white space character is replaced with a space character (#x20) in the normalized value.
```

Comment:
It would be nice to add a reference to the specification, as well as convert this into doc comment on the method on `Attribute`